docs/autogen.py

```diff
@@ -268,204 +51,31 @@ def add_np_implementation(function, docstring):
-                     for method in methods]))
-            blocks.append('\n'.join(subblocks))
-
-        methods = read_page_data(page_data, 'methods')
-
-        for method in methods:
-            blocks.append(render_function(method, method=True))
-
-        functions = read_page_data(page_data, 'functions')
-
-        for function in functions:
-            blocks.append(render_function(function, method=False))
-
-        if not blocks:
-            raise RuntimeError('Found no content for page ' +
-                               page_data['page'])
+def preprocess_docstring(docstring, function, signature):
+    if "backend" in signature and "{{np_implementation}}" in docstring:
+        docstring = add_np_implementation(function, docstring)
+    return docstring
 
-        mkdown = '\n----\n\n'.join(blocks)
-        # Save module page.
-        # Either insert content into existing page,
-        # or create page otherwise.
-        page_name = page_data['page']
-        path = os.path.join(sources_dir, page_name)
-        if os.path.exists(path):
-            template = read_file(path)
-            if '{{autogenerated}}' not in template:
-                raise RuntimeError('Template found for ' + path +
-                                   ' but missing {{autogenerated}}'
-                                   ' tag.')
-            mkdown = template.replace('{{autogenerated}}', mkdown)
-            print('...inserting autogenerated content into template:', path)
-        else:
-            print('...creating new page with autogenerated content:', path)
-        subdir = os.path.dirname(path)
-        if not os.path.exists(subdir):
-            os.makedirs(subdir)
-        with open(path, 'w', encoding='utf-8') as f:
-            f.write(mkdown)
 
-    shutil.copyfile(os.path.join(str(keras_dir), 'CONTRIBUTING.md'),
-                    os.path.join(str(sources_dir), 'contributing.md'))
-    copy_examples(os.path.join(str(keras_dir), 'examples'),
-                  os.path.join(str(sources_dir), 'examples'))
+def generate(dest_dir):
+    template_dir = keras_dir / "docs" / "templates"
+    keras_autodoc.generate(
+        dest_dir,
+        template_dir,
+        PAGES,
+        'https://github.com/keras-team/keras/blob/master',
+        keras_dir / "examples",
+        EXCLUDE,
+        clean_module_name=clean_module_name,
+        post_process_signature=post_process_signature,
+        preprocess_docstring=preprocess_docstring,
+    )
+    readme = (keras_dir / "README.md").read_text()
+    index = (template_dir / "index.md").read_text()
+    index = index.replace("{{autogenerated}}", readme[readme.find("##"):])
+    (dest_dir / "index.md").write_text(index, encoding="utf-8")
+    shutil.copyfile(keras_dir / "CONTRIBUTING.md", dest_dir / "contributing.md")
 
 
 if __name__ == '__main__':
-    generate(os.path.join(str(keras_dir), 'docs', 'sources'))
+    generate(keras_dir / 'docs' / 'sources')
```

Comment:
Please be consistent with `"` and `'` usage; note that we use `'` in general in the codebase

Reply:
Done